lib/intercept.js

```diff
@@ -365,75 +368,105 @@ function activeMocks() {
   return [].concat(...interceptorScopes().map(scope => scope.activeMocks()))
 }
 
-function activate() {
-  if (originalClientRequest) {
-    throw new Error('Nock already active')
-  }
+function getRequestOptionsFromFetchRequest(fetchRequest) {
+  const requestOptions = {
+    method: fetchRequest.method,
+    headers: {},
+  };
 
-  overrideClientRequest()
+  fetchRequest.headers.forEach((value, key) => {
+    requestOptions.headers[key] = value;
+  });
 
-  // ----- Overriding http.request and https.request:
+  const url = new URL(fetchRequest.url);
 
-  common.overrideRequests(function (proto, overriddenRequest, args) {
-    //  NOTE: overriddenRequest is already bound to its module.
+  const options = {
+    ...requestOptions,
+    hostname: url.hostname,
+    port: url.port,
+    path: url.pathname + url.search,
+    proto: url.protocol === 'https:' ? 'https' : 'http',
+  };
 
-    const { options, callback } = common.normalizeClientRequestArgs(...args)
-
-    if (Object.keys(options).length === 0) {
-      // As weird as it is, it's possible to call `http.request` without
-      // options, and it makes a request to localhost or somesuch. We should
-      // support it too, for parity. However it doesn't work today, and fixing
-      // it seems low priority. Giving an explicit error is nicer than
-      // crashing with a weird stack trace. `new ClientRequest()`, nock's
-      // other client-facing entry point, makes a similar check.
-      // https://github.com/nock/nock/pull/1386
-      // https://github.com/nock/nock/pull/1440
-      throw Error(
-        'Making a request with empty `options` is not supported in Nock'
-      )
-    }
+  return options;
+}
 
-    // The option per the docs is `protocol`. Its unclear if this line is meant to override that and is misspelled or if
-    // the intend is to explicitly keep track of which module was called using a separate name.
-    // Either way, `proto` is used as the source of truth from here on out.
-    options.proto = proto
+function convertFetchRequestToClientRequest(fetchRequest) {
+  const url = new URL(fetchRequest.url);
+  const options = {
+    method: fetchRequest.method,
+    host: url.hostname,
+    port: url.port || (url.protocol === 'https:' ? 443 : 80),
+    path: url.pathname + url.search,
+    headers: fetchRequest.headers,
```

Comment:
I think Node expects `OutgoingHeaders` here, which is not compatible with the `Headers` instance. You should be fine doing this:

```js
headers: Object.fromEntries(fetchRequest.headers.entries())
```

Reply:
great catch! 🙏

Reply:
Just be cautious about the compilation target here. I'm not sure what's the support range is for Nock, but `.fromEntries()` and `.entries()` methods may not exist on super old versions of ECMAScript and Node.